File: plataforma-lia/src/components/unified-chat/wizard/dispatchMessages.ts
/**
 * Dispatch messages canonical entre frontend wizard e backend graph.
 *
 * Pattern frágil identificado em audit F-4.11: frontend dispatch
 * sendChatMessage(literal_string) e backend parsea via regex em
 * pipeline_template_node. Mudança de copy em qualquer lado quebra
 * silenciosamente.
 *
 * Esta constante é canonical para AMBOS lados. Mirror em:
 *   lia-agent-system/app/domains/job_creation/dispatch_messages.py
 *
 * NÃO MUDAR sem atualizar dispatch_messages.py simultaneamente.
 *
 * Sensor schema-sync (futuro PR): comparar strings TS↔Python (decodificando
 * ambos como UTF-8 e validando equivalência semântica das regex).
 */

export const WIZARD_PIPELINE_TEMPLATE_DISPATCH = {
  /** Recrutador escolheu template via wizard (sem vaga persistida ainda). */
  applyTemplate: (templateId: string): string =>
    `Aplicar template de pipeline ${templateId}`,

  /** Recrutador já aplicou via fetch separado, avisa backend pra seguir. */
  appliedAck: "Template de pipeline aplicado, pode seguir.",

  /** Recrutador optou por usar pipeline padrão da empresa. */
  useDefault: "Usar pipeline padrão da empresa.",
} as const;

File: lia-agent-system/app/domains/job_creation/dispatch_messages.py
"""Dispatch messages canonical entre frontend wizard e backend graph.

Pattern fragil identificado em audit F-4.11: frontend dispatch
sendChatMessage(literal_string) e backend parsea via regex em
pipeline_template_node. Mudanca de copy em qualquer lado quebra silenciosamente.

Esta constante e canonical para AMBOS lados. Mirror em:
  plataforma-lia/src/components/unified-chat/wizard/dispatchMessages.ts

Sensor schema-sync (futuro PR): validar que strings/regex batem TS<->Python.

NAO MUDAR sem atualizar dispatchMessages.ts simultaneamente.
"""

from __future__ import annotations

# ── Literal dispatch templates (frontend envia EXATAMENTE estas strings) ──
# Mantidos para documentacao + casos onde precisamos emitir do backend.
APPLY_TEMPLATE_TEMPLATE = "Aplicar template de pipeline {template_id}"
APPLIED_ACK_MESSAGE = "Template de pipeline aplicado, pode seguir."
USE_DEFAULT_MESSAGE = "Usar pipeline padrao da empresa."

# ── Regex canonical (mesma semantica dos literais acima, robustos a variacoes) ──
# Liberalidade pre-existente em graph.py:
#  - Patterns sao IGNORECASE
#  - Aceitam variacoes ortograficas comuns (acentos, "padrao"/"padrao")
#  - Aceitam re-arranjo de palavras (search, nao match anchorado)
#
# Strings que DEVEM matchar APPLY_TEMPLATE_PATTERN:
#  - "Aplicar template de pipeline <uuid>"
#
# Strings que DEVEM matchar APPLIED_ACK_PATTERN:
#  - "Template de pipeline aplicado, pode seguir."
#  - "Template de pipeline aplicada"  (variacao genero)
#
# Strings que DEVEM matchar USE_DEFAULT_PATTERN:
#  - "Usar pipeline padrao da empresa."
#  - "Pipeline padrao da empresa"
#  - "Pular template"
APPLY_TEMPLATE_PATTERN = (
    r"aplicar.*?template.*?pipeline.*?"
    r"([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})"
)
APPLIED_ACK_PATTERN = r"template.*?pipeline.*?aplicad[oa]"
USE_DEFAULT_PATTERN = (
    r"(pipeline.*?padr[ãa]o.*?empresa"
    r"|usar.*?padr[ãa]o.*?empresa"
    r"|pular.*?template)"
)
